app/main.py

```diff
@@ -1,34 +1,172 @@
+        self.end = end
+        self.is_drowned = is_drowned
+        self.decks = self.create_ship(start, end)
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    @staticmethod
+    def create_ship(start: tuple[int], end: tuple[int]) -> list:
+        if start[0] == end[0] and start[1] < end[1]:
+            return [
+                Deck(start[0], column)
+                for column in range(start[1], end[1] + 1)
+            ]
+        elif start[0] < end[0] and start[1] == end[1]:
+            return [
+                Deck(row, start[1])
+                for row in range(start[0], end[0] + 1)
+            ]
+        elif start[0] == end[0] and start[1] == end[1]:
+            return [Deck(start[0], start[1])]
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def get_deck(self, row: int, column: int) -> Deck:
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                return deck
+
+    def fire(self, row: int, column: int) -> None:
+        self.get_deck(row, column).is_alive = False
+        if not any(deck.is_alive for deck in self.decks):
+            self.is_drowned = True
 
 
 class Battleship:
-    def __init__(self, ships):
-        # Create a dict `self.field`.
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
-
-    def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+    def __init__(self, ships: list[tuple]) -> None:
+        self.field = self.create_field(ships)
+
+        try:
+            self._validate_field()
+        except ValueError as ve:
+            print(ve)
+            exit()
+
+    @staticmethod
+    def create_field(ships: list[tuple]) -> dict:
+        ships_list = []
+        field_dict = {}
+
+        for ship in ships:
+            ships_list.append(Ship(ship[0], ship[1]))
+
+        for ship in ships_list:
+            for deck in ship.decks:
+                field_dict[(deck.row, deck.column)] = ship
+
+        return field_dict
+
+    def fire(self, location: tuple) -> str:
+        if location in self.field.keys():
+            ship = self.field[location]
+            ship.fire(location[0], location[1])
+            if ship.is_drowned:
+                return "Sunk!"
+            return "Hit!"
+        return "Miss!"
+
+    def print_field(self) -> None:
+        field_string = ""
+
+        for row in range(10):
+            for column in range(10):
+                if (row, column) in self.field.keys():
+                    ship = self.field[(row, column)]
```

Comment:
also I see that `(row, column)` used twice, so maybe better to make a variable for it